torch/lib/THD/master_worker/common/CommandChannel.hpp

```diff
@@ -33,6 +34,8 @@ struct MasterCommandChannel {
   std::thread _error_thread;
 
   port_type _port;
+
+  std::vector<std::unique_ptr<std::mutex>> _mutexes;
```

Comment:
Actually, it seems that it's possible to keep non-movable things in a vector as long as you only initialize it. Instead of using `push_back` in a loop just create it like that `std::vector<std::mutex>(world_size)`